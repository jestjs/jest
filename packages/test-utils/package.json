{
  "name": "@jest/test-utils",
  "version": "30.0.0-alpha.7",
  "private": true,
  "main": "./build/index.js",
  "types": "./build/index.d.ts",
  "exports": {
    ".": {
      "types": "./build/index.d.ts",
      "require": "./build/index.js",
      "import": "./build/index.mjs",
      "default": "./build/index.js"
    },
    "./package.json": "./package.json"
  },
  "dependencies": {
    "@jest/globals": "workspace:*",
    "@jest/pattern": "workspace:*",
    "@jest/types": "workspace:*",
    "@types/node": "*",
    "ansi-regex": "^5.0.1",
    "ansi-styles": "^5.0.0",
    "pretty-format": "workspace:*",
    "semver": "^7.5.3"
  },
  "devDependencies": {
    "@types/semver": "^7.1.0"
  },
  "engines": {
    "node": "^16.10.0 || ^18.12.0 || >=20.0.0"
  }
}
