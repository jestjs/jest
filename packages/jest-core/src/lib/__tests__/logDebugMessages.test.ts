/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import type {WriteStream} from 'tty';
import {makeGlobalConfig, makeProjectConfig} from '@jest/test-utils';
import logDebugMessages from '../logDebugMessages';

jest.mock('../../../package.json', () => ({version: 123}));

jest.mock('myRunner', () => ({name: 'My Runner'}), {virtual: true});

const getOutputStream = (resolve: (message: string) => void) =>
  ({
    write(message: string) {
      resolve(message);
    },
  } as WriteStream);

it('prints the jest version', async () => {
  expect.assertions(1);
  const message = await new Promise<string>(resolve => {
    logDebugMessages(
      makeGlobalConfig({watch: true}),
      makeProjectConfig({testRunner: 'myRunner'}),
      getOutputStream(resolve),
    );
  });

  expect(JSON.parse(message).version).toBe(123);
});

it('prints the test framework name', async () => {
  expect.assertions(1);
  const message = await new Promise<string>(resolve => {
    logDebugMessages(
      makeGlobalConfig({watch: true}),
      makeProjectConfig({testRunner: 'myRunner'}),
      getOutputStream(resolve),
    );
  });

  expect(JSON.parse(message).configs.testRunner).toBe('myRunner');
});

it('prints the config object', async () => {
  expect.assertions(1);
  const globalConfig = makeGlobalConfig({
    watch: true,
  });
  const config = makeProjectConfig({
    automock: false,
    rootDir: '/path/to/dir',
    roots: ['path/to/dir/test'],
    testRunner: 'myRunner',
  });
  const message = await new Promise<string>(resolve => {
    logDebugMessages(globalConfig, config, getOutputStream(resolve));
  });
  expect(message).toMatchSnapshot();
});
