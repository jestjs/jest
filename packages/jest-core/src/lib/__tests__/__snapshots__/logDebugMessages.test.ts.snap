// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`prints the config object 1`] = `
"{
  "configs": {
    "automock": false,
    "cache": false,
    "cacheDirectory": "/test_cache_dir/",
    "clearMocks": false,
    "collectCoverageFrom": [
      "src",
      "!public"
    ],
    "coverageDirectory": "coverage",
    "coveragePathIgnorePatterns": [],
    "coverageReporters": [],
    "cwd": "/test_root_dir/",
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "extensionsToTreatAsEsm": [],
    "fakeTimers": {
      "enableGlobally": false
    },
    "forceCoverageMatch": [],
    "globals": {},
    "haste": {},
    "id": "test_name",
    "injectGlobals": true,
    "moduleDirectories": [],
    "moduleFileExtensions": [
      "js"
    ],
    "moduleNameMapper": [],
    "modulePathIgnorePatterns": [],
    "modulePaths": [],
    "openHandlesTimeout": 1000,
    "prettierPath": "prettier",
    "reporters": [
      "default",
      "custom-reporter-1",
      [
        "custom-reporter-2",
        {
          "configValue": true
        }
      ]
    ],
    "resetMocks": false,
    "resetModules": false,
    "restoreMocks": false,
    "rootDir": "/path/to/dir",
    "roots": [
      "path/to/dir/test"
    ],
    "runner": "jest-runner",
    "runtime": "/test_module_loader_path",
    "sandboxInjectedGlobals": [],
    "setupFiles": [],
    "setupFilesAfterEnv": [],
    "skipFilter": false,
    "skipNodeResolution": false,
    "slowTestThreshold": 5,
    "snapshotFormat": {},
    "snapshotSerializers": [],
    "testEnvironment": "node",
    "testEnvironmentOptions": {},
    "testLocationInResults": false,
    "testMatch": [],
    "testPathIgnorePatterns": [],
    "testRegex": [
      "\\\\.test\\\\.js$"
    ],
    "testRunner": "myRunner",
    "testTimeout": 5000,
    "transform": [],
    "transformIgnorePatterns": [],
    "waitNextEventLoopTurnForUnhandledRejectionEvents": false,
    "watchPathIgnorePatterns": []
  },
  "globalConfig": {
    "bail": 0,
    "changedFilesWithAncestor": false,
    "changedSince": "",
    "ci": false,
    "collectCoverage": false,
    "collectCoverageFrom": [],
    "coverageDirectory": "coverage",
    "coverageProvider": "babel",
    "coverageReporters": [],
    "coverageThreshold": {
      "global": {}
    },
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "expand": false,
    "findRelatedTests": false,
    "forceExit": false,
    "json": false,
    "lastCommit": false,
    "listTests": false,
    "logHeapUsage": false,
    "maxConcurrency": 5,
    "maxWorkers": 2,
    "noStackTrace": false,
    "nonFlagArgs": [],
    "notify": false,
    "notifyMode": "failure-change",
    "onlyChanged": false,
    "onlyFailures": false,
    "openHandlesTimeout": 1000,
    "passWithNoTests": false,
    "projects": [],
    "reporters": [],
    "rootDir": "/test_root_dir/",
    "runInBand": false,
    "runTestsByPath": false,
    "seed": 1234,
    "silent": false,
    "skipFilter": false,
    "snapshotFormat": {},
    "testFailureExitCode": 1,
    "testNamePattern": "",
    "testPathPatterns": [],
    "testSequencer": "@jest/test-sequencer",
    "testTimeout": 5000,
    "updateSnapshot": "none",
    "useStderr": false,
    "verbose": false,
    "waitNextEventLoopTurnForUnhandledRejectionEvents": false,
    "watch": true,
    "watchAll": false,
    "watchPlugins": [],
    "watchman": false
  },
  "version": 123
}
"
`;
